### ISSUE TITLE
[V3 JSON] Direct reference to mutable datastore being returned

### ISSUE THREAD

**Issue by Tobotimus** [Cog-Creators/Red-DiscordBot#1853]
# Other bugs
The JSON driver is returning direct references to the driver's cache. This is should not be happening as mutating the returned value will cause the mutation to also occur in the cache. This can be shown by the following:
```py
>>> dict1 = await self.conf.dict()
>>> dict1
{}
>>> dict1["somekey"] = "somevalue"
>>> await self.conf.dict()
{"somekey": "somevalue"}
```

IMO it's the driver's responsibility to return a copy from its `get()` method, not Config's.

**Comment by Tobotimus** [comment]
Just a note:
The example I provided will only occur if `self.conf.dict()` is actually being retrieved from the driver, not from Config's defaults. That is, the config value must be manually `set()` first, then this symptom will occur.

**Comment by Tobotimus** [comment]
There is a similar issue with setting a value and then mutating it afterwards:

```py
>>> dict1 = {}
>>> await self.conf.dict()
{}
>>> await self.conf.dict.set(dict1)
>>> dict1["foo"] = "bar"
>>> await self.conf.dict()
{"foo": "bar"}
```

**Comment by tekulvw** [comment]
Fixed with #1855
